### DISCUSSION TITLE
Cron Jobs With Flask

### DISCUSSION THREAD

**Discussion by LineIndent** [vercel/community#2955] [Help] [answered]
### Summary

I'd like to use cron jobs with a Flask app but I'm having trouble getting the initial setup right (can't find docs on the issue).
Below is a simple code snippet as demo:

The following is a file ```index.py``` located inside a directory called ```api```:
 
```py
from flask import Flask, jsonify
app = Flask(__name__)

class MyApp:
    num = 0

    @classmethod
    def add_one(cls):
        cls.num += 1
        return cls.num

@app.route("/")
def home():
    num = str(MyApp.add_one())
    return num
```

This is the ```vercel.json``` file located on the same level as the ```api``` directory:

```json
{
    "rewrites": [
        {
            "source": "/api/(.*)",
            "destination": "/api/index"
        }
    ],
    "cron": {
        "api/index": {
            "schedule": "*/5 * * * *"
        }
    }
}
```

### Steps to Reproduce

1. Create an ```api``` directory.
2. Inside ```api``` create a file, ```index.py``` with the code provided above. 
3. Outside the ```api``` directory, create a JSON file, ```vercel.json``` with the JSON data provided above.
4. Deploy the app to Vercel.

**Comment by amyegan** [comment, marked as answer]
Hi @LineIndent! It looks like your cron config isn't quite right. You can find more info in the cron jobs documentation: https://vercel.com/docs/cron-jobs#create-a-cron-job

I was able to get it working by changing the `vercel.json` config to
```json
{
  "rewrites": [
    {
      "source": "/api/(.*)",
      "destination": "/api/index"
    }
  ],
  "crons": [
    {
      "path": "/api/index",
      "schedule": "*/5 * * * *"
    }
  ]
}
```

Keep in mind that there are [limits](https://vercel.com/docs/cron-jobs#limits) depending on which account type you use for the project. Hobby accounts are limited to two cron jobs that each run up to once per day. Pro and Enterprise teams have higher job limits with unlimited invocations.

**Comment by LineIndent** [comment]
Hey @amyegan Thanks for pointing out the limits (and config file) - I think that's why I couldn't track the output.
